bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -1142,10 +1148,7 @@ impl<'a> CircuitInputStateRef<'a> {
                     } else if Word::from(200u64) * length > Word::from(step.gas.0) {
                         return Ok(Some(ExecError::CodeStoreOutOfGas));
                     } else {
-                        return Err(Error::UnexpectedExecStepError(
```

Comment:
I think we should keep this UnexpectedExecStepError which is more conservative. (I tried rollback this change and all the test cases still pass)

Reply:
@hero78119  thanks for your try. in my testing, all trace tests can pass, but for `begin_tx_deploy_nonce`* test  in begin_tx.rs fail due to this line.   before make the change, they pass because ` return Ok(None)` when hit "RETURN" opcode. after this pr's change, success cases will hit L1145.  let me know if you got different result.

Reply:
Ic. how about this way, by just move this explicitly whitelist condition as case 3
```rust
               // case 3 
                if call.is_create() && call.is_success() && step.op == OpcodeId::RETURN {
                return Ok(None);
            }
```
and need to add condition here to mark CREATE/CREATE2 call `is_success` or not
https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/bus-mapping/src/circuit_input_builder/transaction.rs#L50-L65

With that to avoid adding a `Ok(None)` to capture all other cases. It also separate the success conditions with failed condition clearly in codebase. Besides, explicity success is always better, and left other cases is unknown error, wdyt~?

Reply:
> Ic. how about this way, by just move this explicitly whitelist condition as case 3
> 
> ```rust
>                // case 3 
>                 if call.is_create() && call.is_success && step.op == OpcodeId::RETURN {
>                 return Ok(None);
>             }
> ```

sound good to me, will have a try !!

Reply:
updated in [93b2a24](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1275/commits/93b2a247623ee962ea185f3a2ea7b5008380eccc)  thanks for the suggestion!

Reply:
> Update: I think `is_success` is handled correct for create/create2 op: https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/bus-mapping/src/circuit_input_builder/transaction.rs#L60-L63

that is correct !